Makefile

```diff
@@ -380,6 +389,30 @@ stamps/z3-make: stamps/z3-configure
 	$(MAKE) -C $(Z3_BUILD_DIR) install
 	touch $@
 
+$(Z3_BINARY):
+	wget -O "$@" $(Z3_BINARY_URL)/$@
+
+stamps/z3-binary: $(Z3_BINARY) | stamps
+	unzip -qqo $<
+	mkdir -p $(S2E_PREFIX)
+	mkdir -p $(S2E_PREFIX)/include
+	mkdir -p $(S2E_PREFIX)/lib
+	mkdir -p $(S2E_PREFIX)/bin
+	cp -r $(Z3_BINARY_DIR)/include/* $(S2E_PREFIX)/include/
+	cp $(Z3_BINARY_DIR)/bin/*.a $(S2E_PREFIX)/lib/
+	cp $(Z3_BINARY_DIR)/bin/*.so $(S2E_PREFIX)/lib/
+	cp $(Z3_BINARY_DIR)/bin/z3 $(S2E_PREFIX)/bin/
+	rm -r $(Z3_BINARY_DIR)/*
+	touch $@
+
+ifeq ($(USE_Z3_BINARY),no)
+stamps/z3: stamps/z3-make
```

Comment:
Z3 will still get built because stamps/z3-make is referenced elsewhere as a dependency.

Reply:
You might need to replace these dependencies with `stamps/z3`.

Reply:
Thanks for spotting it; done.